### DISCUSSION TITLE
Does closing a db (with close()) automatically remove a changeListener added via db.addChangeListener?

### DISCUSSION THREAD

**Discussion by natgross** [cofu-app/cbl-dart#556] [Q&A] [answered]
Does a changeListener add via await db.addchangeListener() need to be manually removed (via the listenerToken) before explicitly closing the database via await close(), or does the close() cleanly remove it automatically?

**Comment by blaugold** [comment, marked as answer]
Yes, closing a database automatically cleans up all active listeners, streams, queries and replicators.
